solr/contrib/ltr/README.md

```diff
@@ -1,390 +1,24 @@
-
-
-BONUS: Train an actual machine learning model
-
-1. Download and install [liblinear](https://www.csie.ntu.edu.tw/~cjlin/liblinear/)
-
-2. Change `contrib/ltr/example/config.json` "trainingLibraryLocation" to point to the train directory where you installed liblinear.
-
-3. Extract features, train a reranking model, and deploy it to Solr.
-
-  `cd  contrib/ltr/example`
-
-  `python  train_and_upload_demo_model.py -c config.json`
-
-   This script deploys your features from `config.json` "featuresFile" to Solr.  Then it takes the relevance judged query
-   document pairs of "userQueriesFile" and merges it with the features extracted from Solr into a training
-   file.  That file is used to train a linear model, which is then deployed to Solr for you to rerank results.
-
-4. Search and rerank the results using the trained model
-
-   http://localhost:8983/solr/techproducts/query?indent=on&q=test&wt=json&rq={!ltr%20model=ExampleModel%20reRankDocs=25%20efi.user_query=%27test%27}&fl=price,score,name
-
-# Changes to solrconfig.xml
-```xml
-<config>
-  ...
-
-  <!-- Query parser used to rerank top docs with a provided model -->
-  <queryParser name="ltr" class="org.apache.solr.ltr.search.LTRQParserPlugin" />
-
-  <!--  Transformer that will encode the document features in the response.
-  For each document the transformer will add the features as an extra field
-  in the response. The name of the field will be the the name of the
-  transformer enclosed between brackets (in this case [features]).
-  In order to get the feature vector you will have to
-  specify that you want the field (e.g., fl="*,[features])  -->
-
-  <transformer name="features" class="org.apache.solr.ltr.response.transform.LTRFeatureLoggerTransformerFactory">
-    <str name="fvCacheName">QUERY_DOC_FV</str>
-  </transformer>
-
-  <query>
-    ...
-
-    <!-- Cache for storing and fetching feature vectors -->
-    <cache name="QUERY_DOC_FV"
-      class="solr.search.LRUCache"
-      size="4096"
-      initialSize="2048"
-      autowarmCount="4096"
-      regenerator="solr.search.NoOpRegenerator" />
-  </query>
-
-</config>
-
-```
-
-# Defining Features
-In the learning to rank plugin, you can define features in a feature space
-using standard Solr queries. As an example:
-
-###### features.json
-```json
-[
-{ "name": "isBook",
-  "class": "org.apache.solr.ltr.feature.SolrFeature",
-  "params":{ "fq": ["{!terms f=category}book"] }
-},
-{
-  "name":  "documentRecency",
-  "class": "org.apache.solr.ltr.feature.SolrFeature",
-  "params": {
-      "q": "{!func}recip( ms(NOW,publish_date), 3.16e-11, 1, 1)"
-  }
-},
-{
-  "name":"originalScore",
-  "class":"org.apache.solr.ltr.feature.OriginalScoreFeature",
-  "params":{}
-},
-{
-  "name" : "userTextTitleMatch",
-  "class" : "org.apache.solr.ltr.feature.SolrFeature",
-  "params" : { "q" : "{!field f=title}${user_text}" }
```

Comment:
the ref guide does not mention efis with solr feature and value feature like the readme does here